tool/util/lifecycle.dart

```diff
@@ -1,7 +1,33 @@
 part of dsa.links.repository.util;
 
-fail(String msg, {String out}) async {
-  print("ERROR: ${msg}");
+dbg(String msg) {
+  if (isDebugOn) {
+    var prefix = "[DEBUG]";
+    if (currentLinkBuild != null) {
+      prefix += "[${currentLinkDisplayName}]";
+    }
+    var lines = msg.trim().split("\n");
+    for (var line in lines) {
+      line = line.trim();
+      if (line.isEmpty || line == ".") {
+        continue;
+      }
+
+      print("${prefix} ${line}");
+    }
+  }
+}
+
+fail(String msg, {String out, bool firstLineOnlyDebug: false}) async {
+  if (firstLineOnlyDebug && isDebugOn) {
+    msg = msg.split("\n").first;
+  }
+
+  print("[ERROR] ${msg}");
+
+  if (isOptionOn("test")) {
```

Comment:
This checks for the `--test` option, which is used when we don't want to write any JSON files into the repository.